### PR TITLE
Backport test changes to  ReschedulingTriggerShouldKeepOriginalNextFireTime from main to 3.x

### PR DESCRIPTION
**Pull request by jafin**
EDIT: Improvement to test already done in `main` branch.
This is now just a backport of that change to `3.x` branch

As seen in an unrelated PR and the scheduler test failed, expecting a null value for priorRun, but received one. 
https://github.com/quartznet/quartznet/actions/runs/6061704354/job/16447189544

```cs
::error file=D:\a\quartznet\quartznet\src\Quartz.Tests.Unit\Quartz.Tests.Unit.csproj,title=ReschedulingTriggerShouldKeepOriginalNextFireTime::Expected: null%0D%0A  But was:  2023-09-03 01:40:25.5395791+00:00
     
       Failed ReschedulingTriggerShouldKeepOriginalNextFireTime [216 ms]
       Error Message:
          Expected: null
       But was:  2023-09-03 01:40:25.5395791+00:00
     
       Stack Trace:
          at Quartz.Tests.Unit.SchedulerTest.ReschedulingTriggerShouldKeepOriginalNextFireTime() in /_/src/Quartz.Tests.Unit/SchedulerTest.cs:line 315
        at NUnit.Framework.Internal.TaskAwaitAdapter.GenericAdapter`1.GetResult()
        at NUnit.Framework.Internal.AsyncToSyncAdapter.Await(Func`1 invoke)
        at NUnit.Framework.Internal.Commands.TestMethodCommand.RunTestMethod(TestExecutionContext context)
        at NUnit.Framework.Internal.Commands.TestMethodCommand.Execute(TestExecutionContext context)
        at NUnit.Framework.Internal.Commands.BeforeAndAfterTestCommand.<>c__DisplayClass1_0.<Execute>b__0()
        at NUnit.Framework.Internal.Commands.DelegatingTestCommand.RunTestMethodInThreadAbortSafeZone(TestExecutionContext context, Action action)
     
     1)    at Quartz.Tests.Unit.SchedulerTest.ReschedulingTriggerShouldKeepOriginalNextFireTime() in /_/src/Quartz.Tests.Unit/SchedulerTest.cs:line 315
        at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
        at Quartz.Tests.Unit.SchedulerTest.ReschedulingTriggerShouldKeepOriginalNextFireTime()
```

~~It appears it may be a timing issue where the initial assert expects the trigger to not of fired but in some cases it has. (fast pipeline?)~~
~~This change introduces a padding of 1 second to aim to make the first null check more robust.~~
~~- Reenable Linux run for test.~~

### CODE REVIEW

File: src/Quartz.Tests.Unit/SchedulerTest.cs [hunk 1]
```diff
@@ -303,16 +303,19 @@ public async Task ReschedulingTriggerShouldKeepOriginalNextFireTime()
...
+                .StartAt(DateTime.UtcNow.Add(startAtOffset))
                 .Build();
 
             await scheduler.ScheduleJob(job, trigger);
-
             trigger = (IOperableTrigger) await scheduler.GetTrigger(trigger.Key);
             Assert.That(trigger.GetPreviousFireTimeUtc(), Is.EqualTo(null));
+            
+            // Ensure trigger will fire by waiting for elapsed StartAt time.
+            Thread.Sleep(Convert.ToInt16(startAtOffset.TotalMilliseconds));
```
Comment (lahma) [inline src/Quartz.Tests.Unit/SchedulerTest.cs:318]: `await Task.Delay(startAtOffset);` would read better, but I also see that this has already been improved in `main`, maybe should just port the fixes from there?

Comment (jafin) [reply src/Quartz.Tests.Unit/SchedulerTest.cs:318]: sure, didn't even think to check main... will refresh.

Comment (jafin) [reply src/Quartz.Tests.Unit/SchedulerTest.cs]: Done, backported main test to 3.x

**Comment by lahma** [comment]
Thank you!

**Pull request merged**